### DISCUSSION TITLE
ERC4337 Smart Wallet connection with DApps

### DISCUSSION THREAD

**Discussion by sahanKW** [WalletConnect/walletconnect-monorepo#3376] [Developer Support]
I have faced an issue when using WalletConnect with a Smart Contract Wallet based on ERC4337 Account Abstraction.

When we swap some MATIC for USDT in UniSwap (DApp) we are sending the `UserOperation` with transaction `data`  to the Bundler. The transaction is executed by the `EntryPoint` contract and the Smart Contract calls related to swap are executed as `internal transactions`.

When we submit the `transactionHash` returned by the `UserOperation` to UniSwap, it failed to validate the `transactionHash` by displaying a "Swap Failed" message.

If we do this transaction from an EOA wallet like Metamask there won't be any issues and UniSwap is able to validate the  `transactionHash`

I would like to know what can we do in this scenario to submit the correct `transactionHash` for UniSwap (DApp)

**Comment by Marvella0608** [comment]
I also had

**Reply by EtherEmir2** [reply]
such an interesting, but such a difficult question. I would think about it. I’ll join the discussion)!

**Comment by arthur-zhuk** [comment]
Same issue here.

**Comment by lienhage** [comment]
Same issue

**Comment by teddyBos32** [comment]
Oh, really good update

**Reply by mikhpush** [reply]
same issue

**Reply by teddyBos32** [reply]
+++

**Comment by arthur-zhuk** [comment]
To get around this you can obtain the `transactionHash` coming from the bundler and handle the UI/UX on your own wallet (in the Uniswap scenario). Would be great to have a solution here.

**Comment by stefanodecillis** [comment]
do you have a snippet for connecting wallet (erc4337) to wallet connect? there is no that much of documentation at the moment

**Reply by arthur-zhuk** [reply]
https://github.com/zerodevapp/zerodev-walletconnect-example/tree/main/wallets/react-wallet-v2/src

This is a good example.

**Reply by Kitisparks** [reply]
It seems like I'm not the only one who has encountered this problem. Now I’ll figure it out and look at your example.

**Comment by MirrorJust** [comment]
everything seems to be working

**Comment by HatimPatwa** [comment]
Same Issue, need an update on this  can we somehow get TxHash as UserOp hash is created. That way we can respond back to the Dapp ?
